docs/userGuide/syntax/lists.md

```diff
@@ -109,3 +109,126 @@ first number
+* item 3
+* item 4 {icon="/images/deer.jpg"}
+  
+{icon="fas fa-check"}
+</variable>
+</include>
+
+</div>
+
+<box type="warning">
+
+1. Item-level specifications should only be attached to the list heading or list item, not the list content.
+2. A blank line preceding the content is crucial.
+3. It's important to insert a blank line before list-level specifications.
+4. It is possible to any of the icons supported by MarkBind (e.g., Font Awesome, Octicons, Glyphicons, Images and etc.) as the icon of a list item.
+5. Images will maintain a 1 rem bottom margin.
+</box>
+
+<box type="tip" seamless>
+It is possible to specify icons for nested levels using the icons attribute.
+</box>
+<div id="main-example-markbind">
+
+<include src="codeAndOutput.md" boilerplate >
+<variable name="highlightStyle">markdown</variable>
+<variable name="code">
+* item 1
+* item 2 
+  * item 2.1
+  * item 2.2
+  {icon="x"}
+* item 3
+  
+{icon="fas fa-check"}
+</variable>
+</include>
+
+</div>
+
+<box type="tip" seamless>
+The icon appearance can be customized further a class attribute.
+</box>
+<div id="main-example-markbind">
+
+<include src="codeAndOutput.md" boilerplate >
+<variable name="highlightStyle">markdown</variable>
+<variable name="code">
+* item 1 {icon="fas fa-check" class="text-warning"}
+* item 2 {icon="fas fa-check" class="text-danger"}
+* item 3 
+  
+{icon="fas fa-check" class="text-primary"}
+</variable>
+</include>
+
+</div>
+<box type="tip" seamless>
+
+You can adjust the size of the icon by using the size attribute, which accepts the following options: 's', 'm', 'l', 'xl'.
+</box>
+<div id="main-example-markbind">
+
+<include src="codeAndOutput.md" boilerplate >
+<variable name="highlightStyle">markdown</variable>
+<variable name="code">
+
+* item 1 {icon="/images/deer.jpg" size="xs"}
+* item 1 {icon="/images/deer.jpg" size="s"}
+* item 2 {icon="/images/deer.jpg" size="m"}
+* item 3 {icon="/images/deer.jpg" size="l"}
+* item 4 {icon="/images/deer.jpg" size="xl"}
+</variable>
+</include>
+
+</div>
+
+<box type="tip" seamless>
+
+You can craft a media list using Markdown's heading and paragraph syntax.
+</box>
+<div id="main-example-markbind">
+
+<include src="codeAndOutput.md" boilerplate >
+<variable name="highlightStyle">markdown</variable>
+<variable name="code">
+
+* ### item 1
+  
+   This is the content for item 1. 
+   Here we discuss the key aspects and details related to this specific item. 
+   Information is comprehensive, well-organized, 
+   and presented in a clear, concise manner.  
+* ### item 2  
+
+  For item 2, the content delves deeper into the topic, 
+  providing a robust analysis and extensive insights. 
+  The goal is to offer a thorough understanding of the subject matter.
+
+* ### item 3 
+
+  The content for item 3 takes a different angle, offering fresh perspectives and innovative ideas. 
+  The objective is to stimulate thought and inspire creative solutions.
+
+
+{icon="/images/deer.jpg" size="xl" class="rounded"}
```

Comment:
I've modified the sizing to accept pixels (px) as I believe the attribute 'size' is more appropriate. This is because we are adjusting not only the width but also the height.

Reply:
Should we be allowing specifying width and height separately?

16:9 images for example cannot be compressed into 1:1 nicely.

Naming nit: `icon-size/width/height/class` instead to make the attributes from this feature clear too. (e.g. not the list item class, but the icon's class)

Reply:
related: https://github.com/MarkBind/markbind/pull/2316#discussion_r1252088583

Reply:
> Naming nit: icon-size/width/height/class instead to make the attributes from this feature clear too. (e.g. not the list item class, but the icon's class)

@SPWwj if you missed this.

Would recommend `icon-class` at least since `class=` normally would apply to the list item. ([e.g.](https://github.com/MarkBind/markbind/pull/2316#discussion_r1264699066))

But not too sure if having a shorter form for `size/width/height` would be worth it over the consistency and ambiguity (`icon-size/width/height`), something we can get @damithc's user input on.

Reply:
I'm fine to have a `icon-` prefix (or `i-`) for names to make it clear we are styling the icon.

Reply:
Yes, I see the conflict with the default attributes and have already updated to the `i-` naming convention.